src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```diff
@@ -1029,7 +1029,27 @@ public void onShowProfiler(OpenProfileEvent event)
       
       // create new profiler 
       ensureVisible(true);
-      server_.newDocument(
+
+      if (event.getDocId() != null)
+      {
+         server_.getSourceDocument(event.getDocId(), new ServerRequestCallback<SourceDocument>()
+         {
+            @Override
+            public void onResponseReceived(SourceDocument response)
+            {
+               addTab(response);
+            }
+            
+            @Override
+            public void onError(ServerError error)
+            {
+               Debug.logError(error);
```

Comment:
Changed. I wasn't showing this error since I wanted to believe that it was unlikely to have a document failing to be created or retrieved. But if this ever happens, users will get a prompt.